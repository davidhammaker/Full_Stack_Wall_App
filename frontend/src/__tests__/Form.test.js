import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Form from '../components/form';


configure({ adapter: new Adapter() });


it('displays nothing when not logged in', () => {
  const form = shallow(<Form />);
  expect(form.contains(<></>)).toBe(true);
})

it('displays form when logged in', () => {
  const form = renderer.create(
    <Form username='username' token='token' />
  );
  let tree = form.toJSON();
  expect(tree).toMatchSnapshot();
});

