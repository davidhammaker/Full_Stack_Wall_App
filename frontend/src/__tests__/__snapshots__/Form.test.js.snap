// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays form when logged in 1`] = `
<form
  className="pl-4 pr-4 pb-4"
  onSubmit={[Function]}
>
  <small />
  <div
    className="form-group"
  >
    <label>
      New Post:
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="d-flex justify-content-end"
  >
    <input
      className="btn btn-info mt-2"
      type="submit"
      value="Post"
    />
  </div>
</form>
`;
